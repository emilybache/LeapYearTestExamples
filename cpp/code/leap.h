#ifndef APPROVALTESTS_CPP_STARTERPROJECT_LEAP_H
#define APPROVALTESTS_CPP_STARTERPROJECT_LEAP_H

bool isLeap(int y){
    if (y % 400 == 0)
    {
        return true;
    }
    if (y % 100 == 0)
    {
        return false;
    }
    return y % 4 == 0;
}


#endif //APPROVALTESTS_CPP_STARTERPROJECT_LEAP_H
