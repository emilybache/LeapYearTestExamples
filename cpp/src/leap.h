#ifndef LEAP_H
#define LEAP_H

bool isLeap(int y);


#endif
