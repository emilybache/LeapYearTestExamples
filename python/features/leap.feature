Feature: Leap years

  Scenario: Common Year
    When I have a common year
    Then it is not a leap year

  Scenario Outline:
    Given I have year <year>
    When I calculate whether it is a leap year
    Then it should be <result>

    Examples: Common Years
    | year  | result |
    | 1993  | False  |
    | 1900  | False  |

    Examples: Leap Years
    | year  | result |
    | 1992  | True  |
    | 2000  | True  |