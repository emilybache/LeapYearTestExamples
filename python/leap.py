def is_leap(year):
    if year % 400 == 0:
        return True
    if year % 100 == 0:
        return False
    return year % 4 == 0
