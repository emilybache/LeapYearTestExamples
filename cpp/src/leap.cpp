

#include "leap.h"

bool isLeap(int y) {
    if (y % 400 == 0) {
        return true;
    }
    if (y % 100 == 0) {
        return false;
    }
    return y % 4 == 0;
}

