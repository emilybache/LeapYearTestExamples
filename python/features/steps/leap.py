"""See https://behave.readthedocs.io/en/latest/

To run these tests, first install 'behave'.
In the project folder (one above this one) run:

  behave

You should see output that includes this:

    1 feature passed, 0 failed, 0 skipped
    5 scenarios passed, 0 failed, 0 skipped
    14 steps passed, 0 failed, 0 skipped, 0 undefined
"""

import os

from behave import when, then, given

import sys
sys.path.insert(0, os.getcwd())

from leap import is_leap


@when("I have the year {year}")
def step_impl(context, year):
    context.year = int(year)


@then("it is not a leap year")
def step_impl(context):
    assert not is_leap(context.year)


@then("it is a leap year")
def step_impl(context):
    assert is_leap(context.year)


@when("I calculate whether it is a leap year")
def step_impl(context):
    context.is_leap = str(is_leap(context.year))


@then("it should be {result}")
def step_impl(context, result):
    assert context.is_leap == result


@given("I have the year {year}")
def step_impl(context, year):
    context.year = int(year)
