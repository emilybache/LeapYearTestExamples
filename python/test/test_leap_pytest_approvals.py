import pytest
from approvaltests import verify_all_combinations, verify_all, verify, get_scenario_namer

from leap import is_leap


def test_leap_with_combinations():
    verify_all_combinations(is_leap, [[1993, 1992, 1900, 2000]])


def test_leap_with_list():
    results = [(year, is_leap(year)) for year in [1993, 1992, 1900, 2000]]
    verify_all("(year, is leap year)", results)


@pytest.mark.parametrize("year", [1993, 1992, 1900, 2000])
def test_scenarios(year):
    verify("is Leap " + str(year) + ": " + str(is_leap(year)), namer=get_scenario_namer(year))
