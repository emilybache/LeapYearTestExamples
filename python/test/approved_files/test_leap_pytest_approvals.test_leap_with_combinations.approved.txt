args: (1993,) => False
args: (1992,) => True
args: (1900,) => False
args: (2000,) => True
